packages/patterns/src/patterns/patternMatchers.js

```diff
@@ -249,6 +249,14 @@ const makePatternKit = () => {
    * @returns {boolean}
    */
   const checkKind = (specimen, kind, check) => {
+    // check null and undefined as Keys
+    const key =
+      // eslint-disable-next-line no-nested-ternary
+      kind === 'null' ? null : kind === 'undefined' ? undefined : kind;
+    if (key == null) {
```

Comment:
With `@jessie-check` I get

<img width="590" alt="image" src="https://github.com/endojs/endo/assets/273868/646ec040-9410-4ad4-be2b-ea9763d45e83">

as I should. Also, my subjective experience is that I always misread this first as `=== null` and think "What about `undefined`?" Only then do I backtrack and look harder. This has happened to me already on three different occasions just looking at this line of code in this PR. Twice on github. Once in vscode.

We're not about to turn `@jessie-check` on for this file anytime soon because of other violations. But we should not gratuitously violate Jessie rules either. Nor should we make Jessie rules more complicated by allowing this weird exception.

Reply:
cc @mhofman who I know has a contrary view and may want to contribute to this review conversation.

@mhofman ,
I do understand that it is a widespread JS convention. And if one gets used to it (as I have never been able to do), that it is just as safe. But: From the rest of our code, my eyes are trained under the *assumption* that there are no `==` in our code, which is why I always find myself doing this double take.

Reply:
Well you could always bypass it in this case by changing the check to
```suggestion
    if (key !== kind) {
```

But in my experience, `== null` is well understood by most JS programmer.

Reply:
The view from Jessie gives an interesting counterpoint: Jessie is simple enough to be teachable to non-JS programmers. For them, saying "There is no `==` in Jessie" is a lot simpler than making the special case.

Reply:
Not that it would affect things here, but I'd like to avoid **all** comparisons of a variable to itself because `name === name` and `name !== name` give counterintuitive results when the value is `NaN` (and to that effect, I just opened #1631).

However, I did replace this check in https://github.com/endojs/endo/pull/1605/commits/6a77c8873b25e2f5262369aa6453245ba4c03a36 with use of a `singletonKinds` Map\<Kind, unknown> containing only `'null' => null, 'undefined' => undefined`.